### PR TITLE
James 2479

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by trantienduchn** [comment]
failed test

```
[db559e378fc159bf8dadb7be90b542678366cfa7] Failed tests: 
[db559e378fc159bf8dadb7be90b542678366cfa7]   ICSAttachmentWorkflowTest.mailShouldNotContainCalendarContentInTextBodyButAttachment:788 
[db559e378fc159bf8dadb7be90b542678366cfa7] Expecting:
[db559e378fc159bf8dadb7be90b542678366cfa7]  <"* 1 FETCH (FLAGS (\Recent \Seen) BODY[] {1730}
[db559e378fc159bf8dadb7be90b542678366cfa7] Return-Path: <fromUser@james.org>
[db559e378fc159bf8dadb7be90b542678366cfa7] MIME-Version: 1.0
[db559e378fc159bf8dadb7be90b542678366cfa7] Content-Type: multipart/mixed; 
[db559e378fc159bf8dadb7be90b542678366cfa7] 	boundary="----=_Part_72_284003125.1532536290988"
[db559e378fc159bf8dadb7be90b542678366cfa7] Received: from localhost (HELO james.org) ([127.0.0.1])
[db559e378fc159bf8dadb7be90b542678366cfa7]           by 3ce314084f22 (JAMES SMTP Server ) with SMTP ID 1197328949
[db559e378fc159bf8dadb7be90b542678366cfa7]           for <touser@james.org>;
[db559e378fc159bf8dadb7be90b542678366cfa7]           Wed, 25 Jul 2018 16:31:30 +0000 (UTC)
[db559e378fc159bf8dadb7be90b542678366cfa7] Delivered-To: to@localhost.com
[db559e378fc159bf8dadb7be90b542678366cfa7] From: User From <fromUser@localhost.com>
[db559e378fc159bf8dadb7be90b542678366cfa7] Message-ID: <2d0fa16c-4c37-58de-1acc-50b5f6a23a60@localhost.com>
[db559e378fc159bf8dadb7be90b542678366cfa7] To: User To <toUser@linagora.com>
[db559e378fc159bf8dadb7be90b542678366cfa7] Date: Wed, 26 Jul 2017 10:34:02 +0200
[db559e378fc159bf8dadb7be90b542678366cfa7] Subject: Any subject
[db559e378fc159bf8dadb7be90b542678366cfa7] 
[db559e378fc159bf8dadb7be90b542678366cfa7] ------=_Part_72_284003125.1532536290988
[db559e378fc159bf8dadb7be90b542678366cfa7] BEGIN:VCALENDAR
[db559e378fc159bf8dadb7be90b542678366cfa7] PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
[db559e378fc159bf8dadb7be90b542678366cfa7] VERSION:2.0
[db559e378fc159bf8dadb7be90b542678366cfa7] METHOD:REPLY
[db559e378fc159bf8dadb7be90b542678366cfa7] BEGIN:VEVENT
[db559e378fc159bf8dadb7be90b542678366cfa7] CREATED:20170726T081145Z
[db559e378fc159bf8dadb7be90b542678366cfa7] LAST-MODIFIED:20170726T083402Z
[db559e378fc159bf8dadb7be90b542678366cfa7] DTSTAMP:20170726T083402Z
[db559e378fc159bf8dadb7be90b542678366cfa7] UID:f1514f44bf39311568d6407249cb76c48103fcd1fb398c3c501cb72b2d293f36e04
[db559e378fc159bf8dadb7be90b542678366cfa7]  b2aab16e43439a608f28671ab7c10e754cbc85ddee4a07fa8cf8f4f7af05bfb502b8f6
[db559e378fc159bf8dadb7be90b542678366cfa7] Content-class: urn:content-classes:calendarmessage
[db559e378fc159bf8dadb7be90b542678366cfa7] Content-Type: text/calendar; method=REPLY; charset=UTF-8
[db559e378fc159bf8dadb7be90b542678366cfa7] Content-transfer-encoding: 8BIT
[db559e378fc159bf8dadb7be90b542678366cfa7] Content-Disposition: attachment
[db559e378fc159bf8dadb7be90b542678366cfa7] 
[db559e378fc159bf8dadb7be90b542678366cfa7] SUMMARY:[A CONFIRMER] Call OVH
[db559e378fc159bf8dadb7be90b542678366cfa7] PRIORITY:5
[db559e378fc159bf8dadb7be90b542678366cfa7] ORGANIZER;CN=User;X-OBM-ID=id:mailto:userTo@localhost.com
[db559e378fc159bf8dadb7be90b542678366cfa7] ATTENDEE;CN=User;PARTSTAT=ACCEPTED;CUTYPE=INDIVIDUAL;X-OBM-ID=810:
[db559e378fc159bf8dadb7be90b542678366cfa7]  mailto:user1To@linagora.com
[db559e378fc159bf8dadb7be90b542678366cfa7] DTSTART:20170727T130000Z
[db559e378fc159bf8dadb7be90b542678366cfa7] DURATION:PT1H
[db559e378fc159bf8dadb7be90b542678366cfa7] TRANSP:OPAQUE
[db559e378fc159bf8dadb7be90b542678366cfa7] SEQUENCE:2
[db559e378fc159bf8dadb7be90b542678366cfa7] X-LIC-ERROR;X-LIC-ERRORTYPE=VALUE-PARSE-ERROR:No value for X property. Rem
[db559e378fc159bf8dadb7be90b542678366cfa7]  oving entire property:
[db559e378fc159bf8dadb7be90b542678366cfa7] CLASS:PUBLIC
[db559e378fc159bf8dadb7be90b542678366cfa7] X-OBM-DOMAIN:linagora.com
[db559e378fc159bf8dadb7be90b542678366cfa7] X-OBM-DOMAIN-UUID:uuid_value
[db559e378fc159bf8dadb7be90b542678366cfa7] LOCATION:01.78.14.42.61 / 1586
[db559e378fc159bf8dadb7be90b542678366cfa7] X-OBM-ALERT;X-OBM-ID=486:600
[db559e378fc159bf8dadb7be90b542678366cfa7] END:VEVENT
[db559e378fc159bf8dadb7be90b542678366cfa7] END:VCALENDAR
[db559e378fc159bf8dadb7be90b542678366cfa7] 
[db559e378fc159bf8dadb7be90b542678366cfa7] 
[db559e378fc159bf8dadb7be90b542678366cfa7] ------=_Part_72_284003125.1532536290988--
[db559e378fc159bf8dadb7be90b542678366cfa7] )
[db559e378fc159bf8dadb7be90b542678366cfa7] AAAE OK FETCH completed.
[db559e378fc159bf8dadb7be90b542678366cfa7] ">
[db559e378fc159bf8dadb7be90b542678366cfa7] to contain sequence:
[db559e378fc159bf8dadb7be90b542678366cfa7]  <["Content-Type: multipart/mixed", "Content-Disposition: attachment"]>
```

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/UpdateMessagePatchValidatorTest.java [hunk 1]
```diff
@@ -80,8 +81,9 @@ public void validateShouldReturnANonEmptyResultWhenParsingThrows() throws IOExce
         ObjectNode emptyRootNode = new ObjectMapper().createObjectNode();
 
         ObjectMapper mapper = mock(ObjectMapper.class);
+        JsonGenerator jsonGenerator = null;
         when(mapper.readValue(anyString(), eq(UpdateMessagePatch.class)))
-            .thenThrow(new JsonMappingException("Exception when parsing"));
+            .thenThrow(JsonMappingException.from(jsonGenerator, "Exception when parsing"));
```
[the remark this answers is not in the document]
Comment (mbaechler) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/UpdateMessagePatchValidatorTest.java]: you may but AFAIK declaring a null variable is a common practice to the team

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java [hunk 2]
```diff
@@ -146,9 +147,10 @@ public void addShouldNotPropagateExceptionWhenExceptionOccurs() throws Exception
         
         when(messageToElasticSearchJson.convertToJson(eq(message), eq(users)))
             .thenThrow(JsonProcessingException.class);
-        
+
+        JsonGenerator jsonGenerator = null;
         when(messageToElasticSearchJson.convertToJsonWithoutAttachment(eq(message), eq(users)))
-            .thenThrow(new JsonGenerationException("expected error"));
+            .thenThrow(new JsonGenerationException("expected error", jsonGenerator));
```
[the remark this answers is not in the document]
Comment (mbaechler) [reply mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndexTest.java]: you may but AFAIK declaring a null variable is a common practice to the team

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/UpdateMessagePatchValidatorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbaechler) [reply server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/UpdateMessagePatchValidatorTest.java]: if it's important to you, please open the debate with the team

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java [hunk 3]
```diff
@@ -198,38 +196,22 @@ public void deserializerShouldThrowWhenNodeIsNotAnObject() throws Exception {
...
-        JsonNode secondNode = mock(JsonNode.class);
-        when(secondNode.asText())
-            .thenReturn("second");
-        JsonNode thirdNode = mock(JsonNode.class);
-        when(thirdNode.asText())
-            .thenReturn("third");
-        when(mainNode.elements())
-            .thenReturn(ImmutableList.of(firstNode, secondNode, thirdNode).iterator());
-        
+        boolean bigDecimalExact = false;
```
[the remark this answers is not in the document]
Comment (mbaechler) [reply mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java]: I don't even know why it's here !

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**